src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,10 +157,27 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            _getSuggestionsTask = GetSuggestionsTaskAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
         }
     }
 
+    private async Task GetSuggestionsTaskAsync(CancellationToken cancellationToken)
+    {
+        var document = this.BaseViewModel.Session.TriggerDocument;
+        _ = document.GetLanguageService<IGoToDefinitionSymbolService>();
+        var editorRenameService = document.GetRequiredLanguageService<IEditorInlineRenameService>();
+        var renameLocations = await this.BaseViewModel.Session.AllRenameLocationsTask.JoinAsync(cancellationToken)
```

Comment:
I feel `IEditorInlineRenameService` could have an additional property to indicate if the language support contruct rename context. 
If it doesn't support construct rename context then we don't need to wait `this.BaseViewModel.Session.AllRenameLocationsTask`